pkg/auth/handler/webapp/panic_middleware.go

```diff
@@ -51,13 +48,16 @@ func (m *PanicMiddleware) Handle(next http.Handler) http.Handler {
 			if e := recover(); e != nil {
 				err := panicutil.MakeError(e)
 				log.PanicValue(m.Logger.Logger, err)
+				apiError := apierrors.AsAPIError(err)
 
 				if !written {
-					data := make(map[string]interface{})
-					baseViewModel := m.BaseViewModel.ViewModel(r, w)
-					baseViewModel.SetError(err)
-					viewmodels.Embed(data, baseViewModel)
-					m.Renderer.RenderHTML(w, r, TemplateWebFatalErrorHTML, data)
+					result := &webapp.Result{
+						RedirectURI:      "/errors/error",
+						NavigationAction: "replace",
+					}
+					// This call cannot fail.
+					_ = m.ErrorCookie.SetNonRecoverableError(result, apiError)
+					result.WriteResponse(w, r)
```

Comment:
I think this could be bad in ux, usually user will try to refresh the same page if he saw an unexpected error.

Reply:
Are you suggesting we should redirect to the same URL, and use recoverable error?

Reply:
🤔 It seems originally we don't redirect and render `fatal_error.html` directly, why we need to change it?

Reply:
Ok, so what we assume here is panic will only occur on non-recoverable error such as programming error, and is impossible to recover by refreshing the page? If it is the case I think it is ok.

Reply:
I/O error could also lead to panic. I changed back to show the error in the original path, the error is also stored in cookie instead of pinned to the query.